common/src/test/java/org/conscrypt/javax/net/ssl/SSLEngineTest.java

```diff
@@ -925,6 +925,184 @@ public void test_SSLEngine_setSSLParameters() throws Exception {
         c.close();
     }
 
+    @Test
+    public void wrapPreconditions() throws Exception {
+        ByteBuffer buffer = ByteBuffer.allocate(10);
+        ByteBuffer[] buffers = new ByteBuffer[] { buffer, buffer, buffer };
+        ByteBuffer[] badBuffers = new ByteBuffer[] { buffer, buffer, null, buffer };
+
+        // Client/server mode not set => IllegalStateException
+        try {
+            newUnconnectedEngine().wrap(buffer, buffer);
```

Comment:
Is fail not added in the first 3 cases on purpose?

Reply:
No,  just careless of me, thanks for catching! Done.